.github/workflows/build_and_upload_on_push_to_dev.yaml

```diff
@@ -0,0 +1,207 @@
+# Deployment workflow is supposed to be dependent on this one.
+
+# 1. At the workflow start four jobs are triggered in parallel (generate version, building all services)
+# 2. A job to build Docker image with Saturn service is triggered once version generated and Saturn artifacts
+#       are built and uploaded
+# 3. A job to build Docker image with Pluto service and Mercury bundle is triggered once version generated and
+#       Pluto/Mercury artifacts are built and uploaded
+
+name: Build & Upload Fairspace Docker images
+
+env:
+  DOCKER_REGISTRY: ghcr.io
+
+on:
+  push:
+    branches:
+      - "dev"
+
+jobs:
+  # A job to generate one shared unique version tag per build cycle for all built artifacts
+  generate-version:
+    runs-on: ubuntu-latest
+    outputs:
+      output1: ${{ steps.version.outputs.snapshot_version }}
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - id: version #TODO: introduce release version generation strategy
+        name: Generating version tag for artifacts (Snapshot only)
+        run: |
+          # EXTRACT VERSION
+          BRANCH=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}
+          echo "Building images from the branch: $BRANCH"
+          VER=$(cat ./VERSION)
+          echo "Building images of version: $VER"     
+
+          # GET DATE AND TIME FOR VERSIONING 
+          DATE=$(date "+%Y%m%d%H%M%S")
```

Comment:
although it is nice and specific, I wonder if we need the date timestamp, it will lead to many different package files. I would just create a 'latest' stamp, which is always the latest build. Introduce the timestamps at a point where we really need this.

Also I would like distinction between dev and release builds. Everything from the release branch should be ready to ship to production, this is not what I expect from a dev image. Let's discuss in the office today how to manage this.

Reply:
good points, let's discuss it in the office

I will play a bit with the latest tag in the meantime